### PR TITLE
[RI2] Updates to installation steps and RA2 references

### PR DESCRIPTION
**Pull request by michaelspedersen**
* Updated installation steps to match recent changed to Kuberef
  - Also modified text slightly to reference Kuberef and current version of BMRA playbooks
* Started adding content to section 2.2
  - Modified table columns slightly to avoid repeating large sections of text
  - Listed requirements for "Kubernetes Nodes" that are handled by HW requirements and configuration done by Kuberef

### CODE REVIEW

File: doc/ref_impl/cntt-ri2/chapters/chapter02.md [hunk 1]
```diff
@@ -22,10 +22,20 @@ interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
 
 ## 2.2 Reference Architecture Specification
 
-| RA2 Section | RA2 Reference  | Description | Requirement for Basic Profile | Requirement for Network Intensive Profile| RI2 Traceability |
-|---|---|---|---|---|---|
-|Kubernetes Nodes|`ra2.ch.001`|Huge Pages|Not required|It must be possible to enable Huge Pages (2048KiB and 1048576KiB) within the Kubernetes Node OS, exposing schedulable resources `hugepages-2Mi` and `hugepages-1Gi`.|_TBC_|
-|||||||
+| RA2 Section | RA2 Reference  | Specification | Requirement | Requirement for Basic Profile | Requirement for Network Intensive Profile | RI2 Traceability |
+|---|---|---|---|---|---|---|
+|Kubernetes Nodes|`ra2.ch.001`|Huge Pages|It must be possible to enable Huge Pages (2048KiB and 1048576KiB) within the Kubernetes Node OS, exposing schedulable resources `hugepages-2Mi` and `hugepages-1Gi`.|Must support|Not required|[4.3.1](./chapter04.md#431-installation-on-bare-metal-infratructure)|
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter02.md:27]: Typo: The Basic and Network Intensive values have been interchanged
```suggestion
|Kubernetes Nodes|`ra2.ch.001`|Huge Pages|It must be possible to enable Huge Pages (2048KiB and 1048576KiB) within the Kubernetes Node OS, exposing schedulable resources `hugepages-2Mi` and `hugepages-1Gi`.|Not required|Must support|[4.3.1](./chapter04.md#431-installation-on-bare-metal-infratructure)|
```

File: doc/ref_impl/cntt-ri2/chapters/chapter02.md [hunk 2]
```diff
@@ -22,10 +22,20 @@ interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
...
+|Kubernetes Nodes|`ra2.ch.002`|SR-IOV Capable NICs|When hosting workloads matching the Network Intensive profile, the physical machines on which the Kubernetes Nodes run must be equipped with NICs that are SR-IOV capable.|Not required|Must support|[3.3](./chapter03.md#33-infrastructure-requirements)|
+|Kubernetes Nodes|`ra2.ch.003`|SR-IOV Virtual Functions|When hosting workloads matching the Network Intensive profile, SR-IOV virtual functions (VFs) must be configured within the Kubernetes Node OS, as the SR-IOV Device Plugin does not manage the creation of these VFs.|Not required|Must support|[4.3.1](./chapter04.md#431-installation-on-bare-metal-infratructure)|
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter02.md:29]: Should it only be that the creation of the VFs in "within the Kubernetes Node OS" as the configuration can be managed by a SR-IOV CNI plugin?
```suggestion
|Kubernetes Nodes|`ra2.ch.003`|SR-IOV Virtual Functions|When hosting workloads matching the Network Intensive profile, SR-IOV virtual functions (VFs) must be created within the Kubernetes Node OS, as the SR-IOV Device Plugin does not manage the creation of these VFs.|Not required|Must support|[4.3.1](./chapter04.md#431-installation-on-bare-metal-infratructure)|
```

File: doc/ref_impl/cntt-ri2/chapters/chapter02.md [hunk 3]
```diff
@@ -22,10 +22,20 @@ interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
 
 ## 2.2 Reference Architecture Specification
 
-| RA2 Section | RA2 Reference  | Description | Requirement for Basic Profile | Requirement for Network Intensive Profile| RI2 Traceability |
-|---|---|---|---|---|---|
-|Kubernetes Nodes|`ra2.ch.001`|Huge Pages|Not required|It must be possible to enable Huge Pages (2048KiB and 1048576KiB) within the Kubernetes Node OS, exposing schedulable resources `hugepages-2Mi` and `hugepages-1Gi`.|_TBC_|
-|||||||
+| RA2 Section | RA2 Reference  | Specification | Requirement | Requirement for Basic Profile | Requirement for Network Intensive Profile | RI2 Traceability |
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter02.md:25]: Could the values under RA2 Section list the actual RA2 Section # as in "4.2 Kubernetes Nodes" and not just "Kubernetes Nodes"?

File: doc/ref_impl/cntt-ri2/chapters/chapter02.md [hunk 4]
```diff
@@ -22,10 +22,20 @@ interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
...
+|Kubernetes Nodes|`ra2.ch.012`|Kubernetes Node RAM Quantity|The Kubernetes Nodes must be equipped with at least 32GB of RAM. Note, extra should be provisioned to cater for any overhead required by the Operating System and any required OS processes such as the container runtime, Kubernetes agents, etc.|Must support|Must support|[3.3](./chapter03.md#33-infrastructure-requirements)|
+|Kubernetes Nodes|`ra2.ch.013`|Physical NIC Quantity|The physical machines on which the Kubernetes Nodes run must be equipped with at least four (4) Network Interface Card (NIC) ports.|Must support|Must support|[3.3](./chapter03.md#33-infrastructure-requirements)|
+|Kubernetes Nodes|`ra2.ch.014`|Physical NIC Speed - Basic Profile|The NIC ports housed in the physical machines on which the Kubernetes Nodes run for workloads matching the Basic Profile must be at least 10Gbps.|Must support|N/A|[3.3](./chapter03.md#33-infrastructure-requirements)|
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter02.md:37]: Typo: the corresponding Network Intensive ra2.ch.015 is missing

File: doc/ref_impl/cntt-ri2/chapters/chapter02.md (hunk 2 above)
Comment (michaelspedersen) [reply doc/ref_impl/cntt-ri2/chapters/chapter02.md:29]: Yeah, I think "created" is a better choice of word here

File: doc/ref_impl/cntt-ri2/chapters/chapter02.md (hunk 4 above)
Comment (michaelspedersen) [reply doc/ref_impl/cntt-ri2/chapters/chapter02.md:37]: Good catch. I was trying to see if they could be combined, but decided not to - I guess I managed to skip the intensive one while I was at it

File: doc/ref_impl/cntt-ri2/chapters/chapter02.md (hunk 1 above)
Comment (michaelspedersen) [reply doc/ref_impl/cntt-ri2/chapters/chapter02.md]: That was definitely incorrect. However, looking at the RA2 `infra.com.cfg.004` it mentions Must/Must, while the RM and the RA2 `i.cap.018` only requires it for the network intensive profile.
I've aligned with the "Kubernetes Node" requirement for now (Must/Must), even though it is likely only a must for network intensive.

**Comment by pgoyal01** [comment]
@michaelspedersen ra2.ch.016 was renumbered as ra2.ch.017 by the merged PR #2279; there were two ra2.ch.015 and the PCIe requirement was renumbered to ra2.ch.016 and the Immutable infrastructure requirement was renumbered to ra2.ch.017.

**Comment by michaelspedersen** [comment]
> @michaelspedersen ra2.ch.016 was renumbered as ra2.ch.017 by the merged PR #2279; there were two ra2.ch.015 and the PCIe requirement was renumbered to ra2.ch.016 and the Immutable infrastructure requirement was renumbered to ra2.ch.017.

Updated the reference to match

**Review by pgoyal01 — APPROVED** [review]

**Review by rihabbanday — APPROVED** [review]

**Comment by rihabbanday** [comment]
@tomkivlin @scottsteinbrueck Could you please review or merge the PR?

**Review by tomkivlin — APPROVED** [review]

**Pull request merged**